src/Composer/ComposerWindow.vala

```diff
@@ -19,18 +19,21 @@
  */
 
 public class Mail.ComposerWindow : Gtk.ApplicationWindow {
+
     public ComposerWindow (Gtk.Window parent) {
+        this.with_headers (parent, null, null, null);
+    }
+
+    public ComposerWindow.with_headers (Gtk.Window parent, string? to, string? cc, string? subject) {
         Object (
             height_request: 600,
             title: _("New Message"),
             transient_for: parent,
             width_request: 680,
             window_position: Gtk.WindowPosition.CENTER_ON_PARENT
         );
-    }
 
-    construct {
```

Comment:
We want to keep with using GObject-style construction here, so if you want to pass variables into the `construct` block from a constructor, you need to create public `{ get; construct; default = null}` properties for `to`, `cc`, and `subject`.

To make each constructor more sane and non-repeating, we probably should take this time to move anything that isn't set in the constructor arguments into the `construct` block like the `width_request`, `window_position`, etc.

Reply:
I'll implemented the rest of the suggestions, but this I'd like to hang on to. My initial implementation used a separate class to group these fields and encapsulate the parsing. I was told not to do it this way, and to create a constructor with the fields split out. And now I feel like I'd be adding in extra verbosity of new properties, for something that's purely as stylistic convention. (I don't think folding the construct block into the primary constructor changes the semantics.) I think it's okay to break convention sometimes for the sake of simplicity.

I'm happy to go back to the separate ComposeRequest class, but at some point this sort of constant rewriting of PRs is too much work to be putting in for free. I need to be trusted to be able to make a certain amount of design choices of my own.

Reply:
I understand if this is beyond the scope of the work you intended to put in. I've submitted a branch [here](https://github.com/vertexian/mail/pull/1) to fix this issue